modules/contacts/modules.php

```diff
@@ -397,6 +397,45 @@ protected function output() {
     }
 }
 
+/**
+ * @subpackage contacts/handler
+ */
+class Hm_Handler_save_contact extends Hm_Handler_Module
+{
+    public function process()
+    {
+        list($success, $form ) = $this->process_form(array('email_address'));
+        if ($success) {
+            $contacts = $this->get('contact_store');
+            $contact_list = $contacts->getAll();
+            $existingEmails = array_column($contact_list, 'email_address');
+
+            $list_mails = array_unique(explode(",", $form['email_address']));
+
+            foreach ($list_mails as $addr) {
+                $addresses = process_address_fld($addr);
+                $newEmails = array_column($addresses, 'email');
+
+                if (!empty($newEmails)) {
+                    $newContacts = array_filter($newEmails, function ($email) use ($existingEmails) {
+                        return !in_array($email, $existingEmails);
```

Comment:
What if you want to add an existing contact not present in Trusted Senders group as trusted sender? Shouldn't it edit the contact and assign to Trusted Senders group?

Reply:
I see what you do but my question remains - if an email coming from the form is not in contacts group - it is added with a group -> Trusted Senders. If an email IS in the contacts list, it is not edited at all - no group 'Trusted Senders' is added for that email. This sounds like a bug to me.